xml/System.IO.Compression/BrotliDecoder.xml

```diff
@@ -114,11 +114,11 @@
         <Parameter Name="bytesWritten" Type="System.Int32" RefType="out" />
       </Parameters>
       <Docs>
-        <param name="source">To be added.</param>
-        <param name="destination">To be added.</param>
-        <param name="bytesWritten">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="source">The byte read-only span containing the compressed data.</param>
+        <param name="destination">When this method returns, a byte span containing the decompressed data.</param>
+        <param name="bytesWritten">The total number of bytes that were written in the <paramref name="destination" />.</param>
+        <summary>Attempts to decompress data that was compressed with the Brotli algorithm.</summary>
+        <returns><see langword="true" /> on success; <see langword="false" /> otherwise.</returns>
         <remarks>To be added.</remarks>
```

Comment:
What do the parameters/destination contain when we return false? Are there any guarantees we provide about the contents of source never changing, and the contents of destination only changing if we return true? Or is the destination overwritten even for false?

Work adding those remarks.

Reply:
> _Are there any guarantees we provide about the contents of source never changing_

`source` is a read-only span, so I'd be really worried if it changed.

> _and the contents of destination only changing if we return true? Or is the destination overwritten even for false?_

Apparently, we can return `false` with a partially decompressed buffer.
See: [corefx/src/Native/AnyOS/brotli/dec/decode.c#L1287](https://github.com/dotnet/corefx/blob/6ad08fec6cd1fcb7348dfe60fda775ca69476a8d/src/Native/AnyOS/brotli/dec/decode.c#L1287)

Here's what I suggest adding to the remarks. Do you agree?:

```suggestion
         <remarks>
           <format type="text/markdown"><![CDATA[

## Remarks

If this method returns `false`, `destination` may be empty or contain partially decompressed data, with `bytesWritten` being zero or greater than zero but less than the expected total.

           ]]></format>
         </remarks>
```

Reply:
> ... with `bytesWritten` being zero or greater than zero but less than the expected total.

If that's the behavior, we should fix it in source (if possible). Can you test it out and verify that bytesWritten is != zero for any case where we return false? That doesn't follow the "try" pattern. Generally, in the "try" pattern, we set the out parameter to default before returning.

For example:
https://github.com/dotnet/corefx/blob/0954a975f972546314bd3ac4eb6e0a6380a85701/src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs#L614-L623

Regardless of the behavior, I wouldn't document it as such. Try returning false means the out parameters contain undefined/meaningless info and shouldn't be relied upon.